Source/ImPlotWidgets/Private/LineFilledGraphWidget.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) Coconut Lizard Limited. All rights reserved.
+
+#include "ImPlotWidgets/Public/LineFilledGraphWidget.h"
+#include "ImGui/Public/ImPlotImplementation.h"
+
+void SFilledGraphWidget::Construct(const FArguments& InArgs)
+{
+	InData = InArgs._InDataArray;
+
+	ChildSlot
+		[
+			SAssignNew(ImGuiCustomWidget, SImGuiCustomWidget)
+			.ContextIndex(InArgs._ContextIndex)
+			.OnImGuiDrawDelegate(this, &SFilledGraphWidget::OnImGuiDraw)
+		];
+}
+
+void SFilledGraphWidget::OnImGuiDraw() const
+{
+	constexpr int32 WindowFlags = ImGuiWindowFlags_NoDecoration
+		| ImGuiWindowFlags_NoBackground
+		| ImGuiWindowFlags_NoSavedSettings;
+
+	constexpr int32 GraphFlags = ImPlotFlags_CanvasOnly | ImPlotFlags_AntiAliased;
+
+	constexpr int32 AxisFlags = ImPlotAxisFlags_NoDecorations;
+
+	const FGeometry& AllottedGeometry = ImGuiCustomWidget.Get()->GetTickSpaceGeometry();
+
+	ImGui::SetNextWindowPos(ImVec2(0, 0), ImGuiCond_Always);
+	ImGui::SetNextWindowSize(ImVec2(AllottedGeometry.GetLocalSize().X, AllottedGeometry.GetLocalSize().Y), ImGuiCond_Always);
+	ImGui::Begin("ImPlot Demo", NULL, WindowFlags);
+
+	// ---------------------------------
+	{
+		TArray<float> XValues;
+		for (float f = 0; f < InData.Num(); ++f)
+		{
+			XValues.Add(f);
+		}
+
+		float xMaxValue = FMath::Max(XValues);
+		float yMaxValue = FMath::Max(InData);
```

Comment:
These two values can be `const float` or `constexpr float` as they don't change - the compiler will usually get this right even without but it helps the next person to read the code know that these values don't change

Reply:
Well, const yes. They can't be constexpr as we can't evaluate them at compile time (constexpr makes a variable a compile time constant) given we're calling functions on arrays that could change at runtime. (And FMath::Max itself would have to be declared as a constexpr function too).